services/web/server/src/simcore_service_webserver/version_control_db.py

```diff
@@ -490,3 +526,46 @@ async def get_snapshot_content(self, repo_id: int, commit_id: int) -> Dict:
                     return snapshot.content
 
         raise NotFoundError(name="snapshot for commit", value=(repo_id, commit_id))
+
+    async def get_workbench_view(self, repo_id: int, commit_id: int) -> Dict[str, Any]:
+        # FIXME: q&d fix!!!
+        # TODO: move to VersionControlRepositoryInternalAPI
+        async with self.engine.acquire() as conn:
+            if (
+                commit := await self.CommitsOrm(conn)
+                .set_filter(repo_id=repo_id, id=commit_id)
+                .fetch("snapshot_checksum")
+            ):
+                repo = (
+                    await self.ReposOrm(conn)
+                    .set_filter(id=repo_id)
+                    .fetch("project_uuid")
+                )
+                assert repo  # nosec
+
+                # if snapshot differs from wcopy, then show working copy
+                wcopy_project_id = await self._fetch_wcopy_project_id(
+                    repo_id, commit_id, conn
+                )
+
+                # FIXME: in general this is wrong. For the moment,
+                # all wcopies except for the repo's main wcopy (i.e. repo.project_uuid) are READ-ONLY
```

Comment:
same here... also I don't know if I understand them...

Reply:
These are all preparations and shall be resolved in next PR, i.e. #2556